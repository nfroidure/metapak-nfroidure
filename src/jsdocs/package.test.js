'use strict';

const assert = require('assert');
const packageTransformer = require('./package');

describe('Package transformer for jsdocs', () => {
  it('should work with an empty package.json', () => {
    assert.deepEqual(
      packageTransformer({
        metapak: {
          data: {
            files: 'yolo.js',
          },
        },
        greenkeeper: {
          ignore: [],
        },
      }),
      {
        metapak: {
          data: {
            files: 'yolo.js',
          },
        },
        devDependencies: {
          'jsdoc-to-markdown': '^3.0.0',
        },
        scripts: {
          doc: 'mkdir -p .readme;' +
            ' echo "# API" > .readme/API.md;' +
            ' jsdoc2md yolo.js >> .readme/API.md',
        },
        greenkeeper: {
          ignore: ['jsdoc-to-markdown'],
        },
      }
    );
  });
});
