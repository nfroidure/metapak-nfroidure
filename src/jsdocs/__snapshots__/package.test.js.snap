// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Package transformer for jsdocs should work with a typescript setup 1`] = `
Object {
  "devDependencies": Object {
    "jsdoc-to-markdown": "^7.1.1",
  },
  "greenkeeper": Object {
    "ignore": Array [
      "jsdoc-to-markdown",
    ],
  },
  "metapak": Object {
    "configs": Array [
      "typescript",
    ],
    "data": Object {
      "distFiles": "yolo.js",
      "files": "yolo.ts",
    },
  },
  "scripts": Object {
    "doc": "echo \\"# API\\" > API.md; jsdoc2md yolo.js >> API.md && git add API.md",
    "precz": "npm run doc",
    "preversion": "npm run doc",
  },
}
`;

exports[`Package transformer for jsdocs should work with an empty package.json 1`] = `
Object {
  "devDependencies": Object {
    "jsdoc-to-markdown": "^7.1.1",
  },
  "greenkeeper": Object {
    "ignore": Array [
      "jsdoc-to-markdown",
    ],
  },
  "metapak": Object {
    "data": Object {
      "files": "yolo.js",
    },
  },
  "scripts": Object {
    "doc": "echo \\"# API\\" > API.md; jsdoc2md yolo.js >> API.md && git add API.md",
    "precz": "npm run doc",
    "preversion": "npm run doc",
  },
}
`;
